Improve: Upgrade to Sparkle 2

rparet opened the pull request:
#### Brief overview of PR changes/additions

Move from Sparkle 1.25 to Sparkle 2.6.4 for Mac updates.

#### Motivation for adding to Mudlet

Stay current with 3rd party dependencies. 

#### Other info (issues closed, discussion etc)

Fixes https://github.com/Mudlet/Mudlet/issues/6665

Moves sparkle directly into mudlet/mudlet and eliminates the third party dependency.

rparet committed:
Improve: Upgrade to Sparkle 2

vadi2 commented:
> It seems entirely possible to cut out the 3rd party dependency

That would be nice, `mixing-cocoa-and-qt` is just there as it was a way I found this to work. macOS isn't my main development platform and that was just how I got this to work.

rparet committed:
remove 3rd party sparkle-glue, pull sparkle into main bundle

rparet commented:
Updated this PR with a more direct approach. Removed `mixing-cocoa-and-qt` and added direct support for Sparkle as an updater.

Also, included `DEV_SPARKLE` as an environment flag to more easily test updater changes in development. Prior to this, it was more or less hardcoded to ptb/release.

rparet committed:
linter fixes

vadi2 reviewed (approved):
Looks solid! Let's give it a spin.

vadi2 commented:
Looks like the packaging step needs more polish, could you look? https://github.com/Mudlet/Mudlet/actions/runs/11596182977/job/32286624135?pr=7349#step:27:161 @rparet

rparet commented:
on it. really odd that it built cleanly on the branch. I'll have an update shortly.

rparet commented:
@vadi2 https://github.com/Mudlet/installers/pull/119 - note that because this involves code signing, I couldn't really test this as thoroughly as I would normally like. Apologies in advance if this takes more than one pass.

vadi2 commented:
That's okay! Sometimes these things don't lend themselves well to local testing as you say.

vadi2 commented:
Looks like it worked great :+1:
